rosbag2/src/rosbag2/impl/sequential_reader_impl.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2__IMPL__SEQUENTIAL_READER_IMPL_HPP_
+#define ROSBAG2__IMPL__SEQUENTIAL_READER_IMPL_HPP_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "rosbag2/sequential_reader.hpp"
+
+namespace rosbag2
+{
+
+class ReaderImpl
```

Comment:
Let's make this LOCAL -> https://github.com/aws-ros-dev/rosbag2/blob/master/rosbag2/include/rosbag2/visibility_control.hpp#L40